app/src/main/java/br/com/orcinus/orca/app/activity/masking/ViewBasedHardwareRoundedCorners.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2024 Orcinus
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See
+ * the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program. If
+ * not, see https://www.gnu.org/licenses.
+ */
+
+package br.com.orcinus.orca.app.activity.masking
+
+import android.os.Build
+import android.view.RoundedCorner
+import android.view.View
+import androidx.annotation.ChecksSdkIntAtLeast
+import androidx.annotation.RequiresApi
+
+/**
+ * [HardwareRoundedCorners] that provides radii through the [view].
+ *
+ * @param view [View] from which information about the corners of the display are retrieved.
+ */
+internal class ViewBasedHardwareRoundedCorners(private val view: View) : HardwareRoundedCorners {
+  @ChecksSdkIntAtLeast(Build.VERSION_CODES.S)
+  override fun areAvailable(): Boolean {
```

Comment:
[`Masker`](https://github.com/orcinusbr/orca-android/blob/e1e6285a9258c37016784d8f9686e9c8e6186a96/app/src/main/java/br/com/orcinus/orca/app/activity/masking/Masker.kt) invariably recovers from [`Float.NaN`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-float/-na-n.html). This method can be removed.